README.md

```diff
@@ -1,12 +1,162 @@
+    )
+
+paragraph = """Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+Integer augue eros, tincidunt vitae eros eu, faucibus tempus risus.
+Donec ullamcorper velit in arcu fermentum faucibus.
+Etiam finibus tortor ac vestibulum dictum. Vestibulum ultricies risus eu lacus luctus pretium.
+Duis congue et nisl eu fringilla. Mauris lorem metus, varius eget ex eget, ultrices suscipit est.
+Integer nunc risus, auctor posuere vehicula id, rutrum et urna.
+Pellentesque gravida, orci id pharetra tempus, nulla neque sagittis elit, condimentum tempor mi velit et urna.
+Fusce faucibus ac libero facilisis commodo. Quisque condimentum suscipit mi.
+Vivamus augue neque, commodo sagittis mollis sed, mollis in sapien.
+Integer cursus et magna nec cursus.
+Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.
+"""
+
+@callback(
+    Output("results", "children"),
+    Output("notification_wrapper", "children", allow_duplicate=True),
+    Input("btn", "n_clicks"),
+    State("socketio", "socketId"),
+    running=[[Output("results", "children"), "", None]],
+    prevent_initial_call=True,
+)
+def display_status(n_clicks, socket_id):
+    if not n_clicks or not socket_id:
+        return no_update, []
+    notify(socket_id, "Sending the question to LoremLM...")
+    time.sleep(1)
+    notify(socket_id, "Streaming answer...")
+
+    for i, word in enumerate(paragraph.replace("\n", " ").split(" ")):
+        emit("stream", " " * bool(i) + word, namespace="/", to=socket_id)
+        time.sleep(0.05)
+
+    notify(socket_id, "Done!", color="green")
+
+    return paragraph, []
+
+clientside_callback(
+    """connected => !connected""",
+    Output("btn", "disabled"),
+    Input("socketio", "connected"),
+)
+
+clientside_callback(
+    """(notification) => {
+        if (!notification) return dash_clientside.no_update
+        return notification
+    }""",
+    Output("notification_wrapper", "children", allow_duplicate=True),
+    Input("socketio", "data-notification"),
+    prevent_initial_call=True,
+)
+
+clientside_callback(
+    """(word, text) => text + word""",
+    Output("results", "children", allow_duplicate=True),
+    Input("socketio", "data-stream"),
+    State("results", "children"),
+    prevent_initial_call=True,
+)
+
+
+if __name__ == '__main__':
+    app.run_server(debug=True, dev_tools_ui=False)
+```
+
+## SocketIO Reference
+
+```
+- id (string; optional):
+    Unique ID to identify this component in Dash callbacks.
+
+- connected (boolean; optional):
+    Whether the client is connected to the websocket - READONLY.
+
+- eventNames (list of strings; optional):
+    Name of Socket.IO events to listen to.
+
+- socketId (string; optional):
+    The socket ID.
+
+- url (string; optional):
+    The socket.io server URL."""
+
+```
+
+The events can be accessed in a clientsside callback via:
+` Input("socketio", "data-eventName")`
+Note that the event needs top trigger a client-side callback.
```

Comment:
It doesn't need to be clientside callbacks. In a lot of cases it will make sense for it to be with clientside callbacks though.
```suggestion
The events can be accessed in a Dash callbacks (client- or server-side) via:
` Input(<id>, "data-<eventName>")`
```

Reply:
Sure. Shall we add something to show what could be done if the server access the event? So far I used your emit a server side action and update on the client side. I did not have the use atm, where the client, or DashSocketIO, did sent something to the server.